src/libs/auth.js

```diff
@@ -134,6 +137,23 @@ authStore.subscribe((state, oldState) => {
   }
 })
 
+export const refreshTerraProfile = () => new Promise(async (resolve, reject) => {
```

Comment:
There's no need to wrap this in an explicit Promise. An `async` function automatically returns a promise that resolves on completion, and any exceptions result in rejection. So it already has the interface that you want.